src/electionguard-ui/ElectionGuard.UI/ViewModels/ViewKeyCeremonyViewModel.cs

```diff
@@ -118,11 +122,20 @@ private void CeremonyPollingTimer_Tick(object? sender, EventArgs e)
 
         _ = Task.Run(async () =>
         {
-            await UpdateGuardiansData();
-            if (IsAdmin || (!IsAdmin && _joinPressed))
+            try
             {
-                await _mediator!.RunKeyCeremony(IsAdmin);
-            }
+                await UpdateGuardiansData();
+                if (IsAdmin || (!IsAdmin && _joinPressed))
+                {
+                    await _mediator!.RunKeyCeremony(IsAdmin);
+                    ErrorMessage = string.Empty;
+                }
+            }
+            catch (Exception ex)
+            {
+                ErrorMessage = ex.Message;
+                _logger.LogError(ex, "Exception in Key Ceremony at {KeyCeremony.State}", KeyCeremony.State);

```

Comment:
should this be `{State}`, or does it not matter, because it's just subbing from `string[] args`?